meteor/server/api/ingest/commit.ts

```diff
@@ -277,14 +283,20 @@ export async function CommitIngestOperation(
 								if (!canRemoveSegment(currentPartInstance, nextPartInstance, segmentId)) {
 									// Protect live segment from being hidden
 									logger.warn(`Cannot hide live segment ${segmentId}, it will be orphaned`)
+									const oldSegment = Segments.findOne(segmentId, {
```

Comment:
Doing a mongo query in a loop like this doesnt feel great.
I know that it will only execute up to twice, but it would still be good to do them in parallel.
Perhaps this could be all at once inside the  `if (orphanHiddenSegmentIds.size) {` block (Line ~319)?

Also, as orphanedHiddenSegmentPropertiesToPreserve is currently an empty object, it would be nice to skip the mongo query when that is the case (eg `Object.keys(orphanedHiddenSegmentPropertiesToPreserve).length > 1 ? Segments.findOne(....) : undefined`)